src/webview/android/kotlin/RustWebViewClient.kt

```diff
@@ -32,6 +33,18 @@ class RustWebViewClient(context: Context): WebViewClient() {
         return shouldOverride(request.url.toString())
     }
 
+    override fun onLoadResource(view: WebView, url: String): Unit {
```

Comment:
From the [docs](https://developer.android.com/reference/android/webkit/WebViewClient#onLoadResource(android.webkit.WebView,%20java.lang.String)): 
> Notify the host application that the WebView will load the resource specified by the given url.

Looks like this function will be called for any resource and not only page navigation